Sources/Commands/SwiftTool.swift

```diff
@@ -696,6 +746,124 @@ extension SwiftTool: BuildPlanDelegate {
     }
 }
 
+extension SwiftTool: BuildSystemDelegate {
+    /// Called in response to requests for new tools.
+    ///
+    /// The client should return an appropriate tool implementation if recognized.
+    public func lookupTool(_ name: String) -> Tool? {
+        return nil
+    }
+    
+    /// Called to report any form of command failure.
+    ///
+    /// This can may be called to report the failure of a command which has
+    /// executed, but may also be used to report the inability of a command to
+    /// run. It is expected to be used by the client in making decisions with
+    /// regard to cancelling the build.
+    public func hadCommandFailure() {
+    }
+    
+    /// Called to report an unassociated diagnostic from the build system.
+    public func handleDiagnostic(_ diagnostic: llbuildSwift.Diagnostic) {
+    }
+
+    /// Called when a command has changed state.
+    ///
+    /// The system guarantees that any commandStart() call will be paired with
+    /// exactly one \see commandFinished() call.
+    public func commandStatusChanged(_ command: Command, kind: CommandStatusKind) {
+    }
+
+    /// Called when a command is preparing to start.
+    ///
+    /// The system guarantees that any commandStart() call will be paired with
+    /// exactly one \see commandFinished() call.
+    public func commandPreparing(_ command: Command) {
+    }
+
+    /// Called when a command has been started.
+    ///
+    /// The system guarantees that any commandStart() call will be paired with
+    /// exactly one \see commandFinished() call.
+    public func commandStarted(_ command: Command) {
+        guard command.shouldShowStatus else { return }
+        print(command.description)
+    }
+    
+    /// Called to allow the delegate to skip commands without cancelling their
+    /// dependents. See llbuild's should_command_start.
+    public func shouldCommandStart(_ command: Command) -> Bool {
+        return true
+    }
+
+    /// Called when a command has been finished.
+    public func commandFinished(_ command: Command, result: CommandResult) {
+    }
+
+    /// Called to report an error during the execution of a command.
+    public func commandHadError(_ command: Command, message: String) {
+        diagnostics.emit(data: LLBuildCommandErrorDiagnostic(message: message))
+    }
+
+    /// Called to report a note during the execution of a command.
+    public func commandHadNote(_ command: Command, message: String) {
+    }
```

Comment:
We probably shouldn't just discard notes, even if they aren't currently used.

Reply:
How do you suggest we report notes for now?